base/singleton.hpp

```diff
@@ -0,0 +1,24 @@
+#pragma once
+
+#include "base/macros.hpp"
+
+namespace base
+{
+// Simple singleton class.
+template <typename T>
+class Singleton final
```

Comment:
This class does not do what it was made for. Suppose you have your class Foo and you want it to be a singleton.
You do something like that:
```
class FooImpl
{
public:
  FooImpl();
  ...;
};

// And then:
using Foo = Singleton<FooImpl>;
```
But I can still do
```
FooImpl a;
FooImpl b;
```